src/plugins/intel_npu/src/compiler_adapter/include/zero_adapter.hpp

```diff
@@ -0,0 +1,165 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <ze_api.h>
+#include <ze_graph_ext.h>
+
+#include <type_traits>
+#include <utility>
+
+#include "intel_npu/icompiler.hpp"
+#include "intel_npu/utils/logger/logger.hpp"
+#include "intel_npu/utils/zero/zero_types.hpp"
+#include "izero_adapter.hpp"
+
+namespace intel_npu {
+
+#define NotSupportQuery(T) (std::is_same<T, ze_graph_dditable_ext_1_2_t>::value)
+
+// ext version == 1.3 && 1.4, support API (pfnQueryNetworkCreate, pfnQueryNetworkDestroy,
+// pfnQueryNetworkGetSupportedLayers)
+#define SupportAPIGraphQueryNetworkV1(T) \
+    (std::is_same<T, ze_graph_dditable_ext_1_3_t>::value || std::is_same<T, ze_graph_dditable_ext_1_4_t>::value)
+
+// ext version >= 1.5, support API (pfnCreate2, pfnQueryNetworkCreate2, pfnQueryContextMemory)
+#define SupportAPIGraphQueryNetworkV2(T) ((!NotSupportQuery(T) && !SupportAPIGraphQueryNetworkV1(T)))
+
+// For ext version >= 1.5, pfnCreate2 api is avaible
+#define NotSupportGraph2(T)                                                                                        \
+    (std::is_same<T, ze_graph_dditable_ext_1_2_t>::value || std::is_same<T, ze_graph_dditable_ext_1_3_t>::value || \
+     std::is_same<T, ze_graph_dditable_ext_1_4_t>::value)
+
+// A bug inside the driver makes the "pfnGraphGetArgumentMetadata" call not safe for use prior to
+// "ze_graph_dditable_ext_1_6_t".
+// See: E#117498
+#define NotSupportArgumentMetadata(T)                                                                              \
+    (std::is_same<T, ze_graph_dditable_ext_1_2_t>::value || std::is_same<T, ze_graph_dditable_ext_1_3_t>::value || \
+     std::is_same<T, ze_graph_dditable_ext_1_4_t>::value || std::is_same<T, ze_graph_dditable_ext_1_5_t>::value)
+
+#define UseCopyForNativeBinary(T)                                                                                  \
+    (std::is_same<T, ze_graph_dditable_ext_1_2_t>::value || std::is_same<T, ze_graph_dditable_ext_1_3_t>::value || \
+     std::is_same<T, ze_graph_dditable_ext_1_4_t>::value || std::is_same<T, ze_graph_dditable_ext_1_5_t>::value || \
+     std::is_same<T, ze_graph_dditable_ext_1_6_t>::value)
+
+/**
+ * Adapter to use CiD through ZeroAPI
+ */
+template <typename TableExtension>
+class ZeroAdapter final : public IZeroAdapter {
```

Comment:
The same class template is defined in another header with different base.
This is not ok as violates ODR.

Reply:
yeah, removed the file from the backend. Also, the IZeroAdapter class will be removed after PR#27095 is merged.